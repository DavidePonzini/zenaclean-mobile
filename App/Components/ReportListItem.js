import React from 'react'
import { Image, StyleSheet, Text, View } from 'react-native'

export default class ReportListItem extends React.Component {
  render () {
    return (
      <View style={styles.container}>
        <Text style={styles.title}>{this.props.item.title}</Text>
        <Text style={styles.descr}>{this.props.item.descr}</Text>
        <Image style={styles.image} source={{ uri: this.props.item.url }} />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  title: {
    fontSize: 30,
    fontWeight: 'bold'
  },
  descr: {

  },
  container: {
    flex: 1,
    borderColor: 'black',
    borderWidth: 1,
    paddingBottom: 5
  },
  image: {
    width: 100,
    height: 100
  }
})
