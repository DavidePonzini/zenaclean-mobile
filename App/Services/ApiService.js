import Secrets from 'react-native-config'

const baseUrl = Secrets.API_URL

const getAddressFromCoords = ({ lat, lng }, cb) => {
  const googleApiUrl = 'https://maps.googleapis.com/maps/api/geocode/json?address='

  fetch(googleApiUrl + lat + ',' + lng + '&key=' + Secrets.GOOGLE_MAPS_API_KEY)
    .then((response) => response.json())
    .then((responseJson) => cb(responseJson))
}

const getMarkers = (cb) => {
  fetch(baseUrl + 'markers.json')
    .then((response) => response.json())
    .then((responseJson) => { cb(responseJson) })
}

const uploadReport = (infoReport) => {
  return fetch(baseUrl + 'markers.json', {
    method: 'POST',
    body: JSON.stringify(infoReport)
  })
}
export default {
  getAddressFromCoords,
  getMarkers,
  uploadReport
}
