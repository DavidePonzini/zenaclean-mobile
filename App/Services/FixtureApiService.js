const SSNlist = ['LNGSVV60T17H509N', 'HTHHLG90R62B112N', 'BRTHZG56P68C982U']
const emailList = ['prova@gmail.com', 'prova@hotmail.it']
export const user1 = {
  email: emailList[0],
  password: '12345678',
  ssn: SSNlist[0],
  id: '5c5afd7c4be7ff1c8d7daacd'
}

const wallets = {}
wallets[user1.id] = 2
export { wallets }
export const fixtureAddress = 'Via del Camoscio'
export const fixtureMarkers = [{ address: fixtureAddress, title: 'Divano abbandonato', latitude: 44.1234, longitude: 8.1234, description: 'Divano abbandonato in via Gramsci', timestamp: '2018-12-09T14:40:35 -02:00', approved_positive: false, approved_negative: false, _id: '5c6580a33a7f647afccd8920', user_id: user1.id, url: 'images/667856780e39f0bbe7407bfdae603f8e', voted_positive: false, voted_negative: false }]
export const fixtureMessageOk = 'Registrazione effettuata con successo!'
export const fixtureMessageNotOk = 'Utente giÃ  esistente'
export const fixtureUserId = 'verylonguserid'
const totVoti = [32, 45, 21]
export let userId = null

const logInUser = (email, password, cb) => {
  return new Promise(resolve => {
    (user1.email === email && user1.password === password) ? resolve({ status: 'ok' }) : resolve({ status: 'no' })
  }).then((res) => {
    if (res.status === 'ok') {
      userId = fixtureUserId
      cb(null, { id: userId, email, eth_address: null })
    } else {
      cb(new Error('login failed'), res)
    }
  })
}

const getAddressFromCoords = ({ lat, lng }, cb) => {
  return new Promise((resolve) => {
    resolve(fixtureAddress)
  }).then((res) => { cb(null, res) })
}

const getMarkers = (neLat, swLat, swLng, neLng, cb) => {
  return new Promise((resolve) => {
    resolve(JSON.stringify(fixtureMarkers))
  }).then((res) => { cb(JSON.parse(res)) })
}

const uploadReport = (infoReport, cb) => {
  return new Promise((resolve) => {
    resolve({ status: 'ok' })
  }).then(() => {
    cb(null)
  })
}

const registerUser = (email, ssn, password, cb) => {
  return new Promise((resolve) => {
    let checked = false
    for (let i = 0; i < SSNlist.length; i++) {
      if (SSNlist[i] === ssn) checked = true
    }
    for (let i = 0; i < emailList.length; i++) {
      if (emailList[i] === email) checked = true
    }
    if (!checked) { resolve(fixtureMessageOk) }
    if (checked) { resolve(fixtureMessageNotOk) }
  }).then((res) => { cb(null, res) })
}

const voteReport = (v, report, cb) => {
  return new Promise((resolve) => {
    let old = totVoti[report]
    let current = totVoti[report] + 1
    if (old < current) resolve({ status: 'ok' })
    else resolve({ status: 'no' })
  }).then((res, current) => {
    if (res.status === 'ok') {
      totVoti[report] = current
      cb(null)
    } else {
      cb(new Error('login failed'), res)
    }
  })
}

const changePassword = (email, oldPassword, newPassword, confirmPassword, cb) => {
  return new Promise((resolve) => {
    if (user1.password === oldPassword && newPassword === confirmPassword) resolve({ status: 'ok' })
    else resolve({ status: 'no' })
  }).then((res) => {
    if (res.status === 'ok') {
      user1.password = newPassword
      cb(null)
    } else {
      cb(new Error('login failed'), res)
    }
  })
}

const isLoggedIn = () => userId != null

const logoutUser = () => {
  userId = null
}
const rehydrateLogin = () => {
  return new Promise((resolve, reject) => {
    if (userId != null) {
      resolve(userId)
    } else {
      reject(new Error('Not logged in'))
    }
  })
}

const getBalance = (eth_addr, cb) => {
  return new Promise(resolve => {
    resolve({ value: wallets[user1.id] })
  }).then(res => {
    cb(res.value)
  })
}

export default {
  getAddressFromCoords,
  getMarkers,
  uploadReport,
  registerUser,
  logInUser,
  changePassword,
  voteReport,
  isLoggedIn,
  logoutUser,
  rehydrateLogin,
  getBalance
}
