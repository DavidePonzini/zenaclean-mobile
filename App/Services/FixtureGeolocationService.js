const fixturePosition = {
  coords: {
    accuracy: 1135,
    altitude: null,
    altitudeAccuracy: null,
    heading: null,
    latitude: 44.4135423,
    longitude: 8.9554223,
    speed: null
  },
  timestamp: 1547641567823
}

const fixtureRegion = {
  latitude: 44.4135423,
  longitude: 8.9554223,
  latitudeDelta: 0.1,
  longitudeDelta: 0.1
}

let lastLocation = fixturePosition
let currentRegion = fixtureRegion

const setCurrentRegion = reg => { currentRegion = reg }
const getCurrentRegion = () => currentRegion

const geolocateOnce = async () => {
  return new Promise(resolve => {
    let pos = fixturePosition
    lastLocation = pos
    resolve(pos)
  })
}

const getLatestPosition = () => lastLocation

const geolocateContinuously = (cb, cbErr, options) => {
  return setInterval(cb, 100)
}

const stopGeolocating = (id) => {
  clearInterval(id)
}

export default {
  geolocateOnce,
  geolocateContinuously,
  stopGeolocating,
  getLatestPosition,
  getCurrentRegion,
  setCurrentRegion
}
