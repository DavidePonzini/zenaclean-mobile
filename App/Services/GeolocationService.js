let lastLocation = null

// initial region
let currentRegion = {
  latitude: 44.4056,
  longitude: 8.9463,
  latitudeDelta: 0.1,
  longitudeDelta: 0.1
}

const setCurrentRegion = reg => { currentRegion = reg }
const getCurrentRegion = () => currentRegion

const geolocateOnce = async () => {
  return new Promise(resolve => {
    navigator.geolocation.getCurrentPosition((pos) => {
      lastLocation = pos
      resolve(pos)
    })
  })
}

const geolocateContinuously = (cb, cbErr, options) => {
  return navigator.geolocation.watchPosition((pos) => {
    cb(pos)
    lastLocation = pos
  }, cbErr, options)
}

const stopGeolocating = (id) => {
  navigator.geolocation.clearWatch(id)
}

const getLatestPosition = () => lastLocation

export default {
  geolocateOnce,
  geolocateContinuously,
  stopGeolocating,
  getLatestPosition,
  getCurrentRegion,
  setCurrentRegion
}
