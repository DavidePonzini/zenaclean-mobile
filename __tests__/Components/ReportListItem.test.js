import React from 'react'
import { fixtureMarkers } from '../../App/Services/FixtureApiService'
import { shallow, configure } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import 'isomorphic-fetch'

import ReportListItem from '../../App/Components/ReportListItem'
configure({ adapter: new Adapter() })

jest.mock('../../App/Services/ApiService', () => require('../../App/Services/FixtureApiService'))
const onPress = jest.fn()

describe('List tests', () => {
  const wrapper = shallow(<ReportListItem marker={fixtureMarkers[0]} onPress={onPress} />)
  it('renders correctly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
  it('navigates when tapped', () => {
    wrapper.find('TouchableOpacity').first().simulate('Press')
    expect(onPress).toHaveBeenCalled()
  })
})
