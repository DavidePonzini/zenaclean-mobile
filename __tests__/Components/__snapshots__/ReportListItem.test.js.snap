// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List tests renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FEFEFE",
      "flex": 1,
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "elevation": 5,
        "margin": 10,
        "padding": 10,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.8,
        "shadowRadius": 2,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 0.9,
          }
        }
      >
        <Text
          accessibilityLabel="title-report"
          style={
            Object {
              "borderBottomColor": "#EAEAEA",
              "borderBottomWidth": 1,
              "fontFamily": "Avenir-Book",
              "fontSize": 26,
              "margin": 10,
              "padding": 5,
              "textAlign": "center",
            }
          }
          testID="title-report"
        >
          Divano abbandonato
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 0.25,
                "padding": 5,
              }
            }
          >
            <Component
              source={
                Object {
                  "testUri": "../../../App/Assets/Images/default_report_image.png",
                }
              }
              style={
                Object {
                  "height": 80,
                  "width": 80,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "flex": 0.75,
                "flexDirection": "column",
              }
            }
          >
            <Text
              style={
                Object {
                  "flex": 0.1,
                  "fontSize": 14,
                }
              }
            >
              Via del Camoscio
            </Text>
            <Text
              accessibilityLabel="description-report"
              style={
                Object {
                  "flex": 0.8,
                  "fontFamily": "Avenir-Book",
                  "fontSize": 14,
                }
              }
              testID="description-report"
            >
              Divano abbandonato in via Gramsci
            </Text>
            <Text
              style={
                Object {
                  "color": "#808080",
                  "flex": 0.1,
                  "fontFamily": "Avenir-Book",
                  "fontSize": 12,
                }
              }
            >
              9 Dicembre 2018 alle 17:40
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderLeftColor": "#EAEAEA",
            "borderLeftWidth": 1,
            "flex": 0.1,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingLeft": 5,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="gray"
          name="chevron-right"
          size={20}
          style={
            Object {
              "flex": 1,
              "textAlign": "center",
            }
          }
        />
      </View>
    </View>
  </TouchableOpacity>
</View>
`;
